yt/data_objects/grid_patch.py

```diff
@@ -173,6 +181,20 @@ def _prepare_grid(self):
         self.ActiveDimensions = h.grid_dimensions[my_ind]
         self.LeftEdge = h.grid_left_edge[my_ind]
         self.RightEdge = h.grid_right_edge[my_ind]
+        # This can be expensive so we allow people to disable this behavior
+        # via a config option
+        if bool(ytcfg.get('yt', 'reconstruct_index')):
```

Comment:
I don't see any appreciable change in my benchmark, but agreed that it's a worthwhile optimiztion. See the latest version.

FWIW here's my benchmark:

```python
import yt

ds = yt.load('ClusterMerger/Data_000100')

ds.index
```